profile/src/main/java/rdx/works/profile/data/repository/MnemonicRepository.kt

```diff
@@ -21,6 +22,8 @@ class MnemonicRepository @Inject constructor(
     @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher
 ) {
 
+    val mnemonicDataStoreChanged = encryptedPreferencesManager.preferencesSize.map { true }
```

Comment:
this is another good example why we have to collect the state of the viewmodel in the composable with collectAsStateWithLifecycle.

If we would have implemented the WalletViewModel this way, we wouldn't need to add this extra flow there which is kinda pointless.
Because when the user switch back to the wallet from settings screen the collection would trigger again.

Reply:
Not sure what you mean here. We do `collectAsStateWithLifecycle` state of a view model in WalletScreen composable, and I don't think that this particular issue has anything to do with a collection. I am happy to hear if you have an idea how we can avoid that flow and detect situation when mnemonics got deleted.

Reply:
I have explained this in the past, that we do not really `collectAsStateWithLifecycle` because we always [start the collection in the viewmodels](https://rdxworks.slack.com/archives/C06R3DF5P8D/p1702998363054699)!

Reply:
Now there is a better solution to solve the security prompts and it can be reused in other similar situations. 

You can remove the a little bit unclear/vague `preferencesSize` flow and thus the `mnemonicDataStoreChangedq.

Then use the `CheckMnemonicIntegrityUseCase` which does exactly what you want to achieve, right?

Here it is how:
```
@Singleton
class CheckMnemonicIntegrityUseCase @Inject constructor(
    private val mnemonicRepository: MnemonicRepository,
    private val getProfileUseCase: GetProfileUseCase,
    private val keystoreManager: KeystoreManager,
    @ApplicationScope private val applicationScope: CoroutineScope
) {

    private val _isMnemonicIntegrityChanged = MutableStateFlow(false)
    val isMnemonicIntegrityChanged = _isMnemonicIntegrityChanged.asStateFlow()

    fun start() {
        applicationScope.launch {
            invoke()
        }
    }

    private suspend fun invoke() {
         ... // rest of code

         if (keyInvalid) {
              ... // rest of code
              _isMnemonicIntegrityChanged.update{ true }
              return 
         }
        
          _isMnemonicIntegrityChanged.update{ false }
    }
```

call `start` method from `onAppToForeground` in the `MainViewModel` and add the `checkMnemonicIntegrityUseCase.isMnemonicIntegrityChanged` in the `GetEntitiesWithSecurityPromptUseCase`.

That's all 👍🏽

Reply:
do we need to mix `applicationScope` in, rather than just add `private val _isMnemonicIntegrityChanged = MutableStateFlow(false)` which will be updated by `invoke`, observe that flow in MainViewModel and keep the rest unchanged? `onAppToForeground` changes in lock screen PR and I am a bit hesitant to do change you described above, because with lock screen changes, foreground checks are not executed every time pp goes to foreground. I think I would need to do more work to expose flow from that particular use case, when lock screen is taken into account

Reply:
> do we need to mix applicationScope in, rather than just add private val _isMnemonicIntegrityChanged = MutableStateFlow(false) which will be updated by invoke, observe that flow in MainViewModel and keep the rest unchanged? 

yeah I think that should work too as long as the `CheckMnemonicIntegrityUseCase` is singleton

> onAppToForeground changes in lock screen PR and I am a bit hesitant to do change you described above, because with lock screen changes, foreground checks are not executed every time pp goes to foreground. 

Yeah I had a quick look at it. There are not changes that affect the result of the above code. You just run the foreground checks when the app is unlocked. Fair enough.

> I think I would need to do more work to expose flow from that particular use case, when lock screen is taken into account

Unless I am missing something you don't need more work to do on this. 
Kinda similar work has be done with `CloudBackupErrorStream` 

If you want we can first merge the lock screen and then test this.

Reply:
A side note regarding the proposed solution above is to rename `CheckMnemonicIntegrityUseCase` because ideally use cases shouldn't hold any state nor should they be singleton.